torchtune/models/qwen2/_tokenizer.py

```diff
@@ -355,29 +356,48 @@ def tokenize_messages(
         mask = []
         for index, message in enumerate(templated_messages):
             tokens = []
+
+            # message header
+            if message.role != "ipython":
```

Comment:
what happens in the original Qwen2 tokenizer if we have a message from a tool return? should we gat against tool call / tool messages?

Reply:
Originally, this logic for Qwen2 existed in the ChatMLTemplate. We don't do anything for tool responses, i guess because Qwen2 didn't explicitly support it